apps/chat/src/components/Common/ApplicationWizard/CodeAppView/CodeEditor.tsx

```diff
@@ -425,8 +428,9 @@ export const CodeEditor = ({ sourcesFolderId, setValue }: Props) => {
                   </SidebarActionButton>
                   <SidebarActionButton
                     handleClick={() => {
-                      setNewFileFolder(undefined);
-                      setNewFileName('');
+                      handleUploadEmptyFile(
+                        getNextDefaultName('New file', rootFiles),
```

Comment:
Let's put `'New file'` it in constant and memoize defaultName
```
const defaultName = useMemo(() => getNextDefaultName('New file', rootFiles), [rootFiles]);
```
and use it here and on the line 453

On the line 453 I would add disabling `disable={!!newFileName}` to not reset to defaultName if you already modifying new file